-1

Tested at: 4ebfca259beb194df7c790c85cd42628da6ee782

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23712/28937/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-27-2300/src/L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryHexLayerBasedImp1.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-27-2300/src/L1Trigger/L1THGCal/plugins/geometries/NullGeometry.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-27-2300/src/L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryHexImp1.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-27-2300/src/L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryImp1.cc 
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-27-2300/src/L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryImp1.cc: In member function 'virtual void HGCalTriggerGeometryImp1::initialize(const edm::ESHandle<CaloGeometry>&)':
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-27-2300/src/L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryImp1.cc:64:17: error: 'HGCEEDetId' was not declared in this scope
                 HGCEEDetId detid(HGCEE, zside, layer, sector, subsector, cell);
                 ^~~~~~~~~~
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-27-2300/src/L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryImp1.cc:68:82: error: 'detid' was not declared in this scope
                 const auto& ret = cells_to_trigger_cells_.insert( std::make_pair(detid, triggerDetid) );
                                                                                  ^~~~~
</pre>

--- a/Geometry/HGCalCommonData/src/HGCalDDDConstants.cc
+++ b/Geometry/HGCalCommonData/src/HGCalDDDConstants.cc
@@ -397,7 +397,7 @@ bool HGCalDDDConstants::isValidTrap(int layer, int ieta, int iphi) const {
 	  (ieta <= (hgpar_->iEtaMinBH_[indx.first]+
 		    hgpar_->lastModule_[indx.first]-
 		    hgpar_->firstModule_[indx.first])) &&
-	  (iphi > 0) && (iphi < hgpar_->nPhiBinBH_[indx.first]));
+	  (iphi > 0) && (iphi <= hgpar_->nPhiBinBH_[indx.first]));
 }
 
 unsigned int HGCalDDDConstants::layers(bool reco) const {

--- a/Geometry/HGCalGeometry/src/HGCalGeometry.cc
+++ b/Geometry/HGCalGeometry/src/HGCalGeometry.cc
@@ -128,6 +128,10 @@ void HGCalGeometry::newCell( const GlobalPoint& f1 ,
       edm::LogVerbatim("HGCalGeom") << "Valid Id [0] " 
 				    << HGCScintillatorDetId(idc);
 #endif
+    } else {
+      edm::LogWarning("HGCalGeom") << "Check " << HGCScintillatorDetId(idc) 
+				   << " from " << HGCScintillatorDetId(detId)
+				   << " ERROR ???";
     }
   } else {
 #ifdef EDM_ML_DEBUG

--- a/L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryImp1.cc
+++ b/L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryImp1.cc
@@ -2,6 +2,7 @@
 
 #include "L1Trigger/L1THGCal/interface/HGCalTriggerGeometryGenericMapping.h"
 #include "DataFormats/ForwardDetId/interface/HGCTriggerDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCEEDetId.h"
 
 #include <vector>
 #include <iostream>

